### PR TITLE
operators: Add base of pin_operator_digest plugin

### PR DESCRIPTION
**Pull request by chmeliik**
* OSBS-8637

The remainder of this plugin will be implemented in the future, once the
supporting code in osbs-client is ready.

Signed-off-by: Adam Cmiel <acmiel@redhat.com>

Maintainers will complete the following section:
- [x] Commit messages are descriptive enough
- [x] "Signed-off-by:" line is present in each commit
- [x] Code coverage from testing does not decrease and new code is covered
- [x] JSON/YAML configuration changes are updated in the relevant schema
- [x] Changes to metadata also update the documentation for the metadata
- [x] Pull request includes link to an osbs-docs PR for user documentation updates
- [x] New feature can be disabled from a configuration file

### CODE REVIEW

File: atomic_reactor/schemas/container.json [hunk 1]
```diff
@@ -109,6 +109,7 @@
               "additionalProperties": false,
               "required": ["repo", "ref"]
             },
+            "operator_manifest": {"$ref": "#/definitions/operator_manifest"},
```
Comment (chmeliik) [inline atomic_reactor/schemas/container.json]: Why does `container.json` have 2 anyOf schemas? Is this the right place for `operator_manifests`?

Comment (twaugh) [reply atomic_reactor/schemas/container.json]: Because either 'go' or 'remote_source' may be used, but not both.

Comment (chmeliik) [reply atomic_reactor/schemas/container.json]: I see, thanks

File: atomic_reactor/plugins/pre_pin_operator_digest.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        """
+        Initialize pin_operator_digests plugin
+
+        :param tasker: ContainerTasker instance
+        :param workflow: DockerBuildWorkflow instance
+        :param replacement_pullspecs: Dict[str, str], computed in orchestrator,
+                                      provided to workers by osbs-client
+        """
+        super(PinOperatorDigestsPlugin, self).__init__(tasker, workflow)
+        # TODO: container.yaml IS available in worker builds, right?
```
Comment (MartinBasti) [inline atomic_reactor/plugins/pre_pin_operator_digest.py]: yes

File: atomic_reactor/plugins/pre_pin_operator_digest.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+            return True
+        else:
+            self.log.info("Not an operator manifest bundle build, skipping plugin")
+            return False
+
+    def run_in_orchestrator(self):
+        try:
+            self.site_config = get_operator_manifests(self.workflow)
+        except KeyError:
+            raise RuntimeError("operator_manifests configuration missing in reactor config map")
```
Comment (MartinBasti) [inline atomic_reactor/plugins/pre_pin_operator_digest.py]: For BW compatibility, we should not fail if config is missing and build is not an operator bundle build (i.e. let it fail with no allowed_registries when building operator bundle)

Comment (chmeliik) [reply atomic_reactor/plugins/pre_pin_operator_digest.py]: If build is not an operator bundle build this plugin should not run at all (see `should_run()`). Is that what you meant?

Comment (chmeliik) [reply atomic_reactor/plugins/pre_pin_operator_digest.py]: Updated tests to make it a bit more obvious

Comment (MartinBasti) [reply atomic_reactor/plugins/pre_pin_operator_digest.py]: oh right, blind me

**Comment by chmeliik** [comment]
Added code to set param for worker build

**Comment by chmeliik** [comment]
Also added worker param to user_params schema

**Comment by MartinBasti** [comment]
`plugins.md` needs also updates

**Comment by chmeliik** [comment]
Oops, name of plugin was supposed to be singular

**Comment by chmeliik** [comment]
Release notes will be added when plugin is finished

**Pull request merged**